lib/all_your_base_cases.rb

```diff
@@ -37,8 +41,51 @@ def error_message
   end
 end
 
+class AllYourBaseCase::PreProcessor
+  def initialize(row:, index:)
+    @row = row
+    @index = index
+  end
+
+  def run
+    row.merge('expected' => expected_value, 'index' => index)
+  end
+
+  private
+
+  attr_reader :row, :index
+
+  def expected_value
+    return row['expected'] unless row['expected'].nil?
```

Comment:
This is headed in the right direction.

An alternate method would be to special case each one based on the description (or the input values.)